api/api.go

```diff
@@ -25,9 +25,9 @@ type PayloadProblems struct {
 	PayloadError
 }
 
-// PayloadSubmission represents metadata about a successful submission.
-type PayloadSubmission struct {
-	*Submission
+// PayloadSubmittedIteration represents metadata about a submission of an iteration.
+type PayloadSubmittedIteration struct {
+	*SubmittedIteration
```

Comment:
Taking a bit closer look, `Iteration` is the payload sent when submitting a new iteration or a local iteration. `Submission` is the already submitted solution which just seems to me like.

Reading the #252, seems like actually `Submission`, should actually be `Solution` since a solution can have more than one iteration and `Submission` actually contains the number of iterations. 

Thoughts? @kytrinyx